web/views.py

```diff
@@ -290,13 +289,9 @@ def get_initial(self):
     def get_success_url(self):
         return reverse("web:studies-list")
 
-    def completed_consent_frame(self, studies, child):
-        return [
-            r.study
-            for r in Response.objects.filter(
-                study__in=studies, child=child, completed_consent_frame=True
-            ).distinct("study_id")
-        ]
+    def studies_without_completed_consent_frame(self, studies, child):
+        query = Q(responses__child=child, responses__completed_consent_frame=True)
+        return studies.exclude(query)
```

Comment:
lovely! looks like this logic required doing things substantially differently, but it's done very cleanly :)